pkg/pricing/aws/convert.go

```diff
@@ -0,0 +1,139 @@
+	return rateCard, nil
+}
+
+func convertAWSPricingToPurserRateCard(region string, awsPricing *Pricing) *models.RateCard {
+	nodePrice, storagePrice := getResourcePricesFromAWSPricing(awsPricing)
+	return &models.RateCard{
+		IsRateCard:    true,
+		CloudProvider: aws,
+		Region:        region,
+		NodePrices:    nodePrice,
+		StoragePrices: storagePrice,
+	}
+}
+
+func getResourcePricesFromAWSPricing(awsPricing *Pricing) ([]*models.NodePrice, []*models.StoragePrice) {
+	var nodePrices []*models.NodePrice
+	var storagePrices []*models.StoragePrice
+	products := awsPricing.Products
+	planList := awsPricing.Terms
+
+	duplicateComputeInstanceChecker := make(map[string]bool)
+	for _, product := range products {
+		if product.ProductFamily == computeInstance {
+			key := product.Sku + product.Attributes.InstanceType + product.Attributes.OperatingSystem
+			if _, isPresent := duplicateComputeInstanceChecker[key]; !isPresent && product.Attributes.PreInstalledSW == na {
+				nodePrice := getComputeInstancePrice(product, planList)
+				if nodePrice != nil {
+					duplicateComputeInstanceChecker[key] = true
+					logrus.Debugf("Node Price: %v", *nodePrice)
+					// TODO: store/update nodePrice in dgraph
+					nodePrices = append(nodePrices, nodePrice)
+				}
+			}
+		} else if product.ProductFamily == storageInstance {
+			storagePrice := getStorageInstancePrice(product, planList)
+			if storagePrice != nil {
+				logrus.Debugf("Storage Price: %v", *storagePrice)
+				// TODO: store/update storagePrice in dgraph
+				storagePrices = append(storagePrices, storagePrice)
+			}
+		}
+	}
+	return nodePrices, storagePrices
+}
+
+func getComputeInstancePrice(product Product, planList PlanList) *models.NodePrice {
+	for _, pricingAttributes := range planList.OnDemand[product.Sku] {
+		for _, pricingData := range pricingAttributes.PriceDimensions {
+			for _, pricePerUnit := range pricingData.PricePerUnit {
+				priceInFloat64, err := strconv.ParseFloat(pricePerUnit, 64)
+				if err != nil {
+					logrus.Errorf("unable to parse string: %s to float. err: %v", pricePerUnit, err)
+					return nil
+				}
+				// Unit of Compute price USD-perHour
+				productXID := product.Attributes.InstanceType + deliminator + product.Attributes.InstanceFamily + deliminator + product.Attributes.OperatingSystem
+				return &models.NodePrice{
+					ID:              dgraph.ID{Xid: productXID},
+					IsNodePrice:     true,
+					InstanceType:    product.Attributes.InstanceType,
+					InstanceFamily:  product.Attributes.InstanceFamily,
+					OperatingSystem: product.Attributes.OperatingSystem,
+					Price:           priceInFloat64,
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func getStorageInstancePrice(product Product, planList PlanList) *models.StoragePrice {
+	for _, pricingAttributes := range planList.OnDemand[product.Sku] {
+		for _, pricingData := range pricingAttributes.PriceDimensions {
+			for _, pricePerUnit := range pricingData.PricePerUnit {
```

Comment:
This is the same code as above. Can we abstract the common code to a separate method?

Reply:
Done